README.md

```diff
@@ -5,3 +5,415 @@ An implementation of [Soutei](http://okmij.org/ftp/papers/Soutei.pdf).
+"`edit` is permitted when the application tells us there's a user (denoted by the _variable_
+`?user`) associated with the request, and the application tells us that that user is identically the
+author of the post".  Observe that `:-` (which I'll pronounce "when") is like an implication in
+logic.  The second rule (if you're allowed to edit you're allowed to read) is just saying that
+`edit` implies `read` (or, since the implication goes right-to-left, perhaps "`read` is implied by
+`edit`" would be better).
+
+Don't worry if this is still confusing.  The example demonstrates several different features of the
+system.  Nonetheless, I hope you agree it's a relatively compact and clean description of a
+not-completely-trivial authorization policy.
+
+## Using Soutei for Authorization
+
+Architecturally, Soutei provides support for implementing a so-called "policy decision point".  Its
+job is to advise an application on whether it should permit a request by determining if it complies
+with its security policy.  It does _not_ provide policy _enforcement_.  This is to say, you ask
+whether to allow access, and you get an answer, but it's still up to you to send the 403 back to the
+client if that answer was "no".
+
+To make use of the policy engine, you formulate a query and send it along.  For instance, when
+implementing the commenting system for your blog platform, you might ask Soutei `may(comment)` (this
+is concrete syntax, in practice you'll assemble the query using library functions).  Of course,
+whether commenting is permitted depends on information that Soutei doesn't have.  In our example,
+that's at least the identity of the commenting user and the author of the post, and probably some
+more information necessary to deduce the ability to read the post.  Soutei only knows about the
+rules you've given it, so where do these extra facts about this particular commenting operation come
+from?  The application provides them as a parameter to the query.  So in our example, the
+application might ask "is commenting permitted? (and by the way, the post has been published, the
+current user is `bob`, the author of the post is `alice`, and `alice` says `bob` is her friend, in
+case that helps you make your decision)".
+
+This may be a rather different style of access control than you're used to, and it requires a bit of
+a shift in perspective.
+
+TODO: Say more here.
+
+## Syntax and Semantics
+
+Here's a rule about activities:
+
+```prolog
+can(?activity) :-
+  want-to(?activity).
+```
+
+It says that an activity is permitted (`can(?activity)`) when we want to do that activity
+(`want-to(?activity)`).  We can see several features of the syntax in this small example.  Rules
+have two parts (a **head** and a **body**), separated by the symbol `:-` (which I tend to pronounce
+"when"), and terminated with a period.  The head of the rule is `can(?activity)`, and the body of
```

Comment:
Might want to drop the parenthetical about pronunciation, since you said it above.

Reply:
I repeated it assuming people might not read the doc top to bottom (which I suppose is an argument for having the paper linked in both places).  Are the occurrences too close together, do you think?

Reply:
Yeah, that's fair.